internal/services/dataprotection/data_protection_backup_policy_postgresql_flexible_server_resource.go

```diff
@@ -0,0 +1,700 @@
+				})
+			}
+		}
+	}
+
+	return results
+}
+
+func flattenBackupPolicyPostgreSQLFlexibleServerRetentionRules(input []backuppolicies.BasePolicyRule) []BackupPolicyPostgreSQLFlexibleServerRetentionRule {
+	results := make([]BackupPolicyPostgreSQLFlexibleServerRetentionRule, 0)
+	var taggingCriterias []backuppolicies.TaggingCriteria
+
+	for _, item := range input {
+		if backupRule, ok := item.(backuppolicies.AzureBackupRule); ok {
+			if trigger, ok := backupRule.Trigger.(backuppolicies.ScheduleBasedTriggerContext); ok {
+				if trigger.TaggingCriteria != nil {
+					taggingCriterias = trigger.TaggingCriteria
+				}
+			}
+		}
+	}
+
+	for _, item := range input {
+		if retentionRule, ok := item.(backuppolicies.AzureRetentionRule); ok {
+			var name string
+			var taggingPriority int
+			var taggingCriteria []BackupPolicyPostgreSQLFlexibleServerCriteria
+
+			if retentionRule.IsDefault == nil || !pointer.From(retentionRule.IsDefault) {
+				name = retentionRule.Name
+
+				for _, criteria := range taggingCriterias {
+					if strings.EqualFold(criteria.TagInfo.TagName, name) {
+						taggingPriority = int(criteria.TaggingPriority)
+						taggingCriteria = flattenBackupPolicyPostgreSQLFlexibleServerBackupCriteria(criteria.Criteria)
+						break
+					}
+				}
+
+				var lifeCycle []BackupPolicyPostgreSQLFlexibleServerLifeCycle
+				if v := retentionRule.Lifecycles; len(v) > 0 {
+					lifeCycle = flattenBackupPolicyPostgreSQLFlexibleServerLifeCycles(v)
+				}
+
+				results = append(results, BackupPolicyPostgreSQLFlexibleServerRetentionRule{
+					Name:      name,
+					Priority:  taggingPriority,
+					Criteria:  taggingCriteria,
+					LifeCycle: lifeCycle,
+				})
+			}
+		}
+	}
+
+	return results
+}
+
+func flattenBackupPolicyPostgreSQLFlexibleServerLifeCycles(input []backuppolicies.SourceLifeCycle) []BackupPolicyPostgreSQLFlexibleServerLifeCycle {
+	results := make([]BackupPolicyPostgreSQLFlexibleServerLifeCycle, 0)
+
+	for _, item := range input {
+		var duration string
+		var dataStoreType string
+
+		if deleteOption, ok := item.DeleteAfter.(backuppolicies.AbsoluteDeleteOption); ok {
+			duration = deleteOption.Duration
+		}
+
+		dataStoreType = string(item.SourceDataStore.DataStoreType)
+
+		results = append(results, BackupPolicyPostgreSQLFlexibleServerLifeCycle{
+			Duration:      duration,
+			DataStoreType: dataStoreType,
+		})
+	}
+
+	return results
+}
+
+func flattenBackupPolicyPostgreSQLFlexibleServerBackupCriteria(input *[]backuppolicies.BackupCriteria) []BackupPolicyPostgreSQLFlexibleServerCriteria {
+	results := make([]BackupPolicyPostgreSQLFlexibleServerCriteria, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, item := range pointer.From(input) {
+		if criteria, ok := item.(backuppolicies.ScheduleBasedBackupCriteria); ok {
+			var absoluteCriteria string
+			if criteria.AbsoluteCriteria != nil && len(pointer.From(criteria.AbsoluteCriteria)) > 0 {
+				absoluteCriteria = string((pointer.From(criteria.AbsoluteCriteria))[0])
```

Comment:
why is `AbsoluteCriteria` designed as a list and we only take the first element?

Reply:
It is also defined like this in other many [existing TF data protection backup policies](https://github.com/hashicorp/terraform-provider-azurerm/tree/main/internal/services/dataprotection) for different Azure services. As they are very similar, so I refer to them for it.